### PR TITLE
provider/aws: Add key_name_prefix argument to aws_key_pair resource

### PR DESCRIPTION
**Pull request by nicksantamaria**
#### What does this PR do?
This PR adds a new argument to the `aws_key_pair` resource called `key_name_prefix`. This behaves in the same way as `name_prefix` seen on several other AWS resources.

#### Any background context you want to provide?
It is worth noting that the current `key_name` argument will auto-generate an identifier if omitted. This was documented incorrectly which lead me down the path of implementing this (documentation now fixed in #9992).

#### What are the relevant tickets?
I could not find an existing issue requesting this enhancement.

#### Checklist

**Acceptance test coverage of new behaviour**
Implemented - see `builtin/providers/aws/resource_aws_key_pair_test.go`

```
$ make testacc TEST=./builtin/providers/aws TESTARGS='-run=testAccCheckAWSKeyPair_namePrefix'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/11/09 23:40:07 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=testAccCheckAWSKeyPair_namePrefix -timeout 120m
PASS
ok  	github.com/nicksantamaria/terraform/builtin/providers/aws	0.019s
```

**Documentation updates**
Implemented - see `website/source/docs/providers/aws/r/key_pair.html.markdown`

**Well-formed Code**
`make fmt` does not show any failures.

### CODE REVIEW

File: builtin/providers/aws/resource_aws_key_pair.go [hunk 1]
```diff
@@ -27,10 +27,24 @@ func resourceAwsKeyPair() *schema.Resource {
...
+			"key_name_prefix": &schema.Schema{
 				Type:     schema.TypeString,
 				Optional: true,
-				Computed: true,
 				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) > 100 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 100 characters, name is limited to 255", k))
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_key_pair.go]: Any reason why you chose 100 as the max length here?

Comment (nicksantamaria) [inline builtin/providers/aws/resource_aws_key_pair.go]: This matches the validation for the `name_prefix` argument for `aws_security_group` resource. I have double-checked and the maximum key pair name is 255 characters.

screen shot 2016-11-10 at 12 30 06 am

Comment (stack72) [inline builtin/providers/aws/resource_aws_key_pair.go]: Got it :)

**Comment by stack72** [comment]
This LGTM! Thanks for the work on this :)

```
% make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSKeyPair_'                                                               ✭
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/11/09 15:26:41 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSKeyPair_ -timeout 120m
=== RUN   TestAccAWSKeyPair_importBasic
--- PASS: TestAccAWSKeyPair_importBasic (53.48s)
=== RUN   TestAccAWSKeyPair_basic
--- PASS: TestAccAWSKeyPair_basic (30.77s)
=== RUN   TestAccAWSKeyPair_generatedName
--- PASS: TestAccAWSKeyPair_generatedName (22.10s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	106.368s
```

**Pull request merged**